src/api/queries.c

```diff
@@ -509,17 +509,32 @@ int api_queries(struct ftl_conn *api)
 						search[j][0] = '%';
 					}
 
+					// Escape any possible underscore in the
+					// search string to ensure it is not
+					// interpreted as a wildcard by SQLite3
+					for(unsigned int i = 0; i < searchlen; i++)
+					{
+						if(search[j][i] == '_')
+						{
+							// Escape underscore
```

Comment:
I agree with your approach in https://github.com/pi-hole/FTL/issues/2549#issuecomment-3027926764. 

ESCAPE the string, but leave it to users to escape the underscore themselves.

It's a shame there is no intuitive single character available to take in user input and, similar to '*' -> '%', converted to underscores to serve as a single character wildcard. The obvious choices each have collisions, '?' as field separator in the URL and '.' in domain names, that rule out anything like that.

Leaving underscore as wildcard but the ability to escape it when needed seems the cleanest approach.